internal/pkg/agent/install/uninstall.go

```diff
@@ -59,6 +60,48 @@ func Uninstall(ctx context.Context, cfgFile, topPath, uninstallToken string, log
 		return fmt.Errorf("uninstall must be run from outside the installed path '%s'", topPath)
 	}
 
+	// check if the agent was installed using --unprivileged by checking the file vault for the agent secret (needed on darwin to correctly load the vault)
+	unprivileged, err := checkForUnprivilegedVault(ctx)
+	if err != nil {
+		return fmt.Errorf("error checking for unprivileged vault: %w", err)
+	}
+
+	// will only notify fleet of the uninstall command if it can gather config and agentinfo, and is not a stand-alone install
+	localFleet := false
+	notifyFleet := false
+	var ai *info.AgentInfo
+	c, err := operations.LoadFullAgentConfig(ctx, log, cfgFile, false, unprivileged)
+	if err != nil {
+		pt.Describe(fmt.Sprintf("unable to read agent config to determine if notifying Fleet is needed: %v", err))
+	}
+	cfg, err := configuration.NewFromConfig(c)
+	if err != nil {
+		pt.Describe(fmt.Sprintf("notify Fleet: unable to transform *config.Config to *configuration.Configuration: %v", err))
+	}
+
+	if cfg != nil && !configuration.IsStandalone(cfg.Fleet) {
+		ai, err = info.NewAgentInfo(ctx, false)
+		if err != nil {
+			pt.Describe(fmt.Sprintf("unable to read agent info, Fleet will not be notified of uninstall: %v", err))
+		} else {
+			notifyFleet = true
+		}
+		if cfg.Fleet != nil && cfg.Fleet.Server != nil {
+			localFleet = true
+		}
+	}
+
+	// Skip on Windows because of https://github.com/elastic/elastic-agent/issues/5952
+	// Once the root-cause is identified then this can be re-enabled on Windows.
+	// Notify fleet-server while it is still running if it's running locally
+	if notifyFleet && localFleet && runtime.GOOS != "windows" {
+		if cfg.Fleet.Client.Transport != nil {
+			cfg.Fleet.Client.Transport.TLS.VerificationMode = tlscommon.VerifyCertificate
+		}
+		cfg.Fleet.Client.Hosts = []string{cfg.Fleet.Client.Host}
```

Comment:
I'm not sure what's going on here; but `Client.Hosts` had the remote (cloud) fleet-server address, and `Client.Host` was the internal port.
I'm going to take some time tomorrow to verify that our normal agent operations still use the internal port when the agent is running fleet-server.

Reply:
agent does connect to `localhost:8221`; but strangely enough fleet-server does not populate the `server.address` attribute in HTTP logs; i'll open a fleet-server PR to fix this

Reply:
`cfg.Fleet.Client.Hosts = []string{cfg.Fleet.Client.Host}` is a bit weird. This is perhaps something else that could be encapsulated into the output of an RPC call to the running agent service since agent itself must know that these represent two different things.

Reply:
We should avoid using an RPC call because we don't want to introduce the requirement that an agent is running for the uninstall to succeed.

Reply:
What I *think* is happening here:
- elastic agent sets `Host` from the [enroll command](https://github.com/elastic/elastic-agent/blob/b07566b96a95daa11897c7fbf0b9320618e2540b/internal/pkg/agent/cmd/enroll_cmd.go#L984) 
- in the policy_change handler elastic-agent uses [only `Hosts` field coming from Fleet](https://github.com/elastic/elastic-agent/blob/8c43ad333837380f1646c3b007ff6339b1593a94/internal/pkg/agent/application/actions/handlers/handler_action_policy_change.go#L202-L209) and tries to reset the other fields. 

I guess that the merged result of that is not quite right in this scenario and we end up with the discrepancy between `Host` and `Hosts`

From discussion with @nchaulet  at the time, it was determined that Fleet will send back only `Hosts` field while the separate `Host`, `Protocol`, `Path` is only handled by agent when initializing config to store command line args.

I filed an [issue](https://github.com/elastic/elastic-agent/issues/4771) to have a look and possible remote Host, Protocol and Path fields from the agent config at the time but maybe this needs further evaluation/discussion taking into account fleet bootstrap scenarios

Reply:
Can we link to that issue and add this explanation to the code? This is going to be really not obvious to someone reading this code without the context of this PR.